### ISSUE TITLE
A11y: [ConfirmModal] Body/description text is not read by screen reader although it can contain crucial information

### ISSUE THREAD

**Issue by L-M-K-B** [grafana/grafana#67297]
### **User Experience:** 
Visually impaired users who depend on screen readers face difficulty in understanding what the modal exactly is about since they don't get the information contained in the body/description.

### **Repro Steps:**

1. Open localhost:3000
2. Navigate to an existing dashboard
3. Add a row by using the _Add_ dropdown in the header
4. Navigate to the _delete_ button in the row header and open the _Delete row_ modal
5. Check what the screen reader tells users when opening the modal and navigating to different buttons

**Actual Result:** 
Body/description is not mentioned by screen reader. 

**Expected Result:**
The text is read by the screen reader to make sure visually impaired users get all available information to make an informed decision.

_Note: We are looking for a more general solution that includes Modal as the parent component._

### **Severity:**
`SEV2`

### **Area:**
`ConfirmModal` (Saga Design System)

**Comment by joshhunt** [comment]
The cause for this probably lies with `<Modal />`, which is where the fix should be.

**Comment by ckbedwell** [comment]
I can't repro this one. My screen reader can navigate and read all the content in the modal just fine. I highlighted a few other issues in the video I found with the modal, though. 

- The aria-label on the yes button is wrong ([I can see this has been fixed now actually](https://github.com/grafana/grafana/issues/66046))
- When using the screen reader shortcuts such as heading navigation it can still list all the content outside the modal and therefore move reading focus to it. This can easily be fixed by adding `aria-hidden="true"` to the app root (or similar container) when the modal is open.

https://user-images.githubusercontent.com/6906380/234641946-8320210e-1933-4f1b-b848-1b89478f92ea.mov

**Comment by ashharrison90** [comment]
i also can't repro this, so going to close 👍 

> When using the screen reader shortcuts such as heading navigation it can still list all the content outside the modal and therefore move reading focus to it. This can easily be fixed by adding aria-hidden="true" to the app root (or similar container) when the modal is open (or adding aria-modal="true" on the modal itself.

@ckbedwell i think this is now fixed as well - `floating-ui` handles this for us by setting `aria-hidden` on content outside the modal/drawer/any other floating element we've specified to be modal 🥳
